pkg/block/metadata/hash.go

```diff
@@ -0,0 +1,100 @@
+// Copyright (c) The Thanos Authors.
+// Licensed under the Apache License 2.0.
+
+package metadata
+
+import (
+	"crypto/sha256"
+	"encoding/hex"
+	"fmt"
+	"io"
+	"os"
+	"path/filepath"
+	"strings"
+
+	"github.com/pkg/errors"
+)
+
+// HashKind indicates what type of hash it is.
+type HashKind string
+
+const (
+	// SHA256Kind shows that SHA256 has been used to generate the hash.
+	SHA256Kind HashKind = "SHA256"
+)
+
+// ObjectHash stores the hash of an object in the object storage.
+type ObjectHash struct {
+	Kind  HashKind
+	Value string
+}
+
+// Equal returns true if two hashes are equal.
+func (oh *ObjectHash) Equal(other *ObjectHash) bool {
+	return oh.Value == other.Value
+}
+
+// CalculateHash calculates the hash of the given type.
+func CalculateHash(p string, k HashKind) (ObjectHash, error) {
+	switch k {
+	case SHA256Kind:
+		f, err := os.Open(p)
+		if err != nil {
+			return ObjectHash{}, errors.Wrap(err, "opening file")
+		}
+		defer f.Close()
+
+		h := sha256.New()
+
+		if _, err := io.Copy(h, f); err != nil {
+			return ObjectHash{}, errors.Wrap(err, "copying")
+		}
+
+		return ObjectHash{
+			Kind:  SHA256Kind,
+			Value: hex.EncodeToString(h.Sum(nil)),
+		}, nil
+	default:
+		panic(fmt.Sprintf("hash kind %v is not supported; fill a bug, please", k))
+	}
+}
+
+// GetHashDir calculates the hashes of the given kind of all files inside of dir.
+// Ignores the given files.
+func GetHashDir(dir string, k HashKind, ignore ...string) (map[string]ObjectHash, error) {
```

Comment:
Given this is `CalculateHash` while iterating over a directory, I'm wondering if we should name this function `CalculateHashDir`